test/unexpected-set.js

```diff
@@ -165,13 +169,46 @@ describe('unexpected-set', () => {
     });
   });
 
+  describe('to have elements satisfying assertion', () => {
+    it('should succeed', () => {
+      expect(new Set([1, 2, 3]), 'to have elements satisfying', [1, 2, 3]);
+    });
+
+    it('should fail with a diff', () => {
+      expect(
+        () => {
+          expect(new Set([1, 2, 'foo']), 'to have elements satisfying', [
+            1,
+            'foo',
+          ]);
+        },
+        'to throw',
+        "expected Set([ 1, 2, 'foo' ]) to have elements satisfying [ 1, 'foo' ]\n" +
+          '\n' +
+          'Set([\n' +
+          '  1,\n' +
+          '  2, // should be removed\n' +
+          "  'foo'\n" +
+          '])'
+      );
+    });
+  });
+
   describe('to satisfy assertion', () => {
     describe('with a Set instance', () => {
       describe('against another Set instance', () => {
         it('should succeed', () => {
           expect(new Set([1, 2]), 'to satisfy', new Set([2, 1]));
         });
 
+        it('should succeed with partial object matches', () => {
+          expect(
+            new Set([{ foo: true, bar: 1 }]),
+            'to satisfy',
+            new Set([{ foo: true }])
+          );
+        });
```

Comment:
Hmm, didn't we decide to make this case fail?

Reply:
Ah, maybe - my understanding was we decided that much like array-like we'd enforce the size being the same but still honour exhaustively for nested comparisons.

That being said, maybe the fact that sets are percieved to be unordered means this is a bad piece of behaviour. One to ponder some more perhaps.

Reply:
What I took away from our conversation was that this exact case was the main thing to fix for `better-shallow-deep-equal` to DTRT, and that it also made more sense to force `Set` `to satisfy <whatever>` into "exhaustively" mode, similar to `array-like`?

Reply:
The primary issue was allowing `<Set> to satisfy <array-like>` - otherwise we'd end up with comparisons succeeding that we just don't want to. A close second was that "to satisfy" allows subsets by default, which is also an issue.

Once we only allow Set to Set and enforce the same number of items (which is what this PR does so far :) ) then we can separately decide how strict we want the element matching to be. Will write about this separately below.